### PR TITLE
Detect single letter variable names enumerated in PEP-0008

### PR DESCRIPTION
**Pull request by mtmiller**
Fix for #341. Could be improved further to include named parameters, maybe other cases I can't think of where names are declared.

### CODE REVIEW

File: pycodestyle.py [hunk 1]
```diff
@@ -1150,6 +1150,37 @@ def comparison_type(logical_line, noqa):
...
 
 
+def ambiguous_identifier(logical_line, tokens):
+    r"""Never use the characters 'l', 'O', or 'I' as variable names.
+
+    In some fonts, these characters are indistinguishable from the numerals
+    one and zero. When tempted to use 'l', use 'L' instead.
+
+    Okay: L = 0
+    E741: l = 0
```
Comment (IanLee1521) [inline pycodestyle.py]: Thanks @mtmiller ! Could you update the doc tests with the rest of the cases that are not allowed?

Also, the 'as' test cases are missing.

**Comment by mtmiller** [comment]
Added checks for names after `class`, `def`, `global`, and `nonlocal`, with new error IDs for function and class definitions versus local variables.

Now reports errors for almost everything in the original example in #341.

Still missing checks for for loop iterators, function parameters, and the comma-separated lists of variables that can appear in assignments, for loops, arg lists, and `global` and `nonlocal` statements.

**Comment by sigmavirus24** [comment]
This will have to be included in 2.1.0 or some other 2.x.0 release.

**Comment by IanLee1521** [comment]
Next version will be 2.1.0, going to go ahead and merge this. Thanks @mtmiller !

**Pull request merged**